### ISSUE TITLE
Bug: require-await should require await for async gen instantiation

### ISSUE THREAD

**Issue by janwirth** [eslint/eslint#19757]
### What rule do you want to change?

require-await

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
test("should throw when trying to use resource without proper disposal", async () => {
    // Start first resource but don't complete the iteration
    const iterator1 = client.useExclusiveResource("resource1");
    const firstValue = await iterator1.next();
    expect(firstValue.value).toEqual({
      status: "exclusive_resource_acquired",
      resourceId: "resource1",
    });

    // Try to use second resource while first is still active
    let error: Error | undefined;
    try {
      // The error will be thrown asynchronously when next() is called
      const iterator2 = await client.useExclusiveResource("resource2");
      await iterator2.next();
    } catch (e) {
      error = e as Error;
    }

    // Should throw because first resource is still active
    expect(error?.message).toBe(
      "Resource resource1 is still active and not disposed"
    );

    // Complete the first resource to clean up
    await iterator1.next();
  });
```

### What does the rule currently do for this code?

Not highlight first resource acquisition call.

### What will the rule do after it's changed?

Highlight the call

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

I'm doing async generators over nodeJS worker threads to unblock the eletron main thread while talking to a pglite.

The generator may throw asynchronously during setup (before the first next()) call.

I'm not sure if this is just bad design but right now the code works, however I should await all my generator setup calls.

Would be great to allow this to be configurable.
